RecoJets/JetProducers/plugins/CSJetProducer.h

```diff
@@ -0,0 +1,53 @@
+#ifndef RecoJets_JetProducers_CSJetProducer_h
+#define RecoJets_JetProducers_CSJetProducer_h
+
+/* *********************************************************
+  \class CSJetProducer
+
+  \brief Jet producer to produce CMS-style constituent subtracted jets
+
+  \author   Marta Verweij
+  \version  
+
+         Notes on implementation:
+
+         Reimplementation of constituent subtraction from fastjet contrib package (1.0.0)
+         to allow the use of eta-dependent rho and rho_m for the constituents 
+         inside a jet
+
+ ************************************************************/
+
+
+#include "RecoJets/JetProducers/plugins/VirtualJetProducer.h"
+
+namespace cms
+{
+  class CSJetProducer : public VirtualJetProducer
+  {
+  public:
+
+    CSJetProducer(const edm::ParameterSet& ps);
+
+    virtual ~CSJetProducer() {}
+
+    virtual void produce( edm::Event & iEvent, const edm::EventSetup & iSetup );
+    
+  protected:
+
+    virtual void runAlgorithm( edm::Event& iEvent, const edm::EventSetup& iSetup );
+
+    static bool function_used_for_sorting(std::pair<double,int> i,std::pair<double, int> j);
+    
+     // calls VirtualJetProducer::inputTowers
+    //virtual void inputTowers();
```

Comment:
Please remove this commented out code, or add another line of comment which explains why you want to keep them